core/src/main/java/org/elasticsearch/search/aggregations/pipeline/bucketscript/BucketScriptPipelineAggregator.java

```diff
@@ -153,22 +161,110 @@ protected void doReadFrom(StreamInput in) throws IOException {
 
     public static class Factory extends PipelineAggregatorFactory {
 
-        private Script script;
-        private final ValueFormatter formatter;
-        private GapPolicy gapPolicy;
-        private Map<String, String> bucketsPathsMap;
+        private final Script script;
+        private final Map<String, String> bucketsPathsMap;
+        private String format = null;
+        private GapPolicy gapPolicy = GapPolicy.SKIP;
```

Comment:
I'm a bit worried about factories holding default values, because then we could run into issues if the client and the server are not on the same version?